addons/binding/org.openhab.binding.robonect/src/main/java/org/openhab/binding/robonect/handler/RobonectHandler.java

```diff
@@ -0,0 +1,401 @@
+                        break;
+                    case CHANNEL_STATUS_MODE:
+                        if (command instanceof StringType) {
+                            setMowerMode(command);
+                        } else {
+                            logger.debug("Got job remote start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_MOWER_STATUS_STARTED:
+                        if (command instanceof OnOffType) {
+                            handleStartStop((OnOffType) command);
+                        } else {
+                            logger.debug("Got stopped update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_REMOTE_START:
+                        if (command instanceof StringType) {
+                            setRemoteStartJobSetting(command);
+                        } else {
+                            logger.debug("Got job remote start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_AFTER_MODE:
+                        if (command instanceof StringType) {
+                            setAfterModeJobSetting(command);
+                        } else {
+                            logger.debug("Got job after mode update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_START:
+                        if (command instanceof StringType) {
+                            setStartJobSetting(command);
+                        } else {
+                            logger.debug("Got job start update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                    case CHANNEL_JOB_END:
+                        if (command instanceof StringType) {
+                            setEndJobSetting(command);
+                        } else {
+                            logger.debug("Got job end update of type {} but StringType is expected.",
+                                    command.getClass().getName());
+                        }
+                        break;
+                }
+
+            }
+            updateStatus(ThingStatus.ONLINE);
+        } catch (RobonectCommunicationException rce) {
+            logger.error("Failed to communicate with the mower. Taking it offline.", rce);
+            updateState(CHANNEL_STATUS, new DecimalType(MowerStatus.OFFLINE.getStatusCode()));
```

Comment:
well, that channel was thought to handle the case where the mower went offline and stays offline for a longer period. Unfortunately openhab/esh does not give access to the thing state in rules yet (there is an open issue about this https://github.com/eclipse/smarthome/issues/1654 ).

However, I will replace this with a trigger channel and a configuration for the timeout on how long the mower is expected to be offline.